src/Symfony/Component/HttpClient/Response/AmpResponse.php

```diff
@@ -0,0 +1,326 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Response;
+
+use Amp\ByteStream\StreamException;
+use Amp\CancellationTokenSource;
+use Amp\Http\Client\HttpClient;
+use Amp\Http\Client\HttpException;
+use Amp\Http\Client\Request;
+use Amp\Http\Client\Response;
+use Amp\Loop;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpClient\Chunk\FirstChunk;
+use Symfony\Component\HttpClient\Exception\InvalidArgumentException;
+use Symfony\Component\HttpClient\Exception\TransportException;
+use Symfony\Component\HttpClient\HttpClientTrait;
+use Symfony\Component\HttpClient\Internal\AmpClientState;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @internal
+ */
+final class AmpResponse implements ResponseInterface
+{
+    use ResponseTrait;
+
+    private $multi;
+    private $client;
+    private $request;
+    private $shouldBuffer;
+    private $canceller;
+
+    /**
+     * @internal
+     */
+    public function __construct(AmpClientState $multi, Request $request, array $options, ?LoggerInterface $logger)
+    {
+        $this->multi = $multi;
+        $this->logger = $logger;
+        $this->timeout = $options['timeout'];
+        $this->info = [
+            'response_headers' => [],
+            'url' => (string) $request->getUri(),
+            'error' => null,
+            'canceled' => false,
+            'http_method' => $request->getMethod(),
+            'http_code' => 0,
+            'redirect_count' => 0,
+            'start_time' => 0.0,
+            'connect_time' => 0.0, // todo
+            'redirect_time' => 0.0,
+            'pretransfer_time' => 0.0, // todo
+            'starttransfer_time' => 0.0, // todo
+            'total_time' => 0.0, // todo
+            'namelookup_time' => 0.0, // todo
+            'size_upload' => 0, // todo
+            'size_download' => 0, // todo
+            'primary_ip' => '', // todo
+            'primary_port' => 0, // todo
+            'user_data' => $options['user_data'],
+            'debug' => '', // todo
+        ];
```

Comment:
See https://github.com/amphp/http-client/blob/4633d5d61bbaeaba3d154c04b4616ccd88a29b62/src/Interceptor/LogHttpArchive.php for inspiration how most of these can be implemented.

Reply:
Thanks, I've added an `AmpListener` to compute the timings + primary IP/port.